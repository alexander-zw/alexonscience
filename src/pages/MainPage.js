import '../styles/MainPage.css';
import hello_there from '../images/hello-there.png';
import general_kenobi from '../images/general-kenobi.png';

function MainPage() {
    return (
        <div>
            <div className="title-div">
                <span className="title">Alexander Wu</span>
            </div>

            <div className="text-div">
                <p>Psst... check out my <a href="https://tinyurl.com/alexonscience">Youtube Channel</a>!</p>
            </div>

            <div id="kenobi-image">
                <img className="bottom" src={general_kenobi} alt="Hello there"/>
                <img className="top" src={hello_there} alt="General Kenobi"/>
            </div>
        </div>
    );
}

export default MainPage;
