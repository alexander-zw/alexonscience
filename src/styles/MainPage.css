/* Google Fonts. */
@import url('https://fonts.googleapis.com/css2?family=Oleo+Script&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Open+Sans&display=swap');

.title-div {
    padding: 1% 3% 1% 3%;
    background-color: #cbeff5;
}

.title {
    font-size: 4em;
    font-family: 'Oleo Script', 'Georgia', serif;
    background-image: linear-gradient(to right, #249141, #249c8a, #079cb3, #3464eb);
    -webkit-background-clip: text;
    -moz-background-clip: text;
    background-clip: text;
    color: transparent;
}

.text-div {
    padding-left: 3%;
    font-family: 'Open Sans', sans-serif;
    font-size: 1.3em;
}

body {
    background-color: #cbf5d8;
}

#kenobi-image img {
    position: absolute;
    width: 50%;
    left: 50%;
    transform: translate(-50%,0);
    background-color: #cbf5d8;
    -webkit-transition: opacity 1s ease-in-out;
    -moz-transition: opacity 1s ease-in-out;
    -o-transition: opacity 1s ease-in-out;
    transition: opacity 1s ease-in-out;
}

#kenobi-image img.top:hover {
    opacity:0;
}
