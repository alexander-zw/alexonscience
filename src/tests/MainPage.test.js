import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MainPage from '../pages/MainPage';

test('renders my name', () => {
    render(<MainPage />);
    const linkElement = screen.getByText(/Alexander Wu/i);
    expect(linkElement).toBeInTheDocument();
});

it('renders correctly', () => {
    const tree = renderer.create(<MainPage />).toJSON();
    expect(tree).toMatchSnapshot();
});
