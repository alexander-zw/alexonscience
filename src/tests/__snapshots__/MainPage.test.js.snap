// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="title-div"
  >
    <span
      className="title"
    >
      Alexander Wu
    </span>
  </div>
  <div
    className="text-div"
  >
    <p>
      Psst... check out my 
      <a
        href="https://tinyurl.com/alexonscience"
      >
        Youtube Channel
      </a>
      !
    </p>
  </div>
  <div
    id="kenobi-image"
  >
    <img
      alt="Hello there"
      className="bottom"
      src="general-kenobi.png"
    />
    <img
      alt="General Kenobi"
      className="top"
      src="hello-there.png"
    />
  </div>
</div>
`;
